src/advanced/time.md

```diff
@@ -0,0 +1,287 @@
+This logic should meet the following criteria:
+
+- **Using validators**. As Exonum is used to power permissioned blockchain,
+  it is natural to assume that the inputs to business logic determining time
+  are supplied by validators. This assumption could be generalized to support
+  abstract semi-trusted identifiable entities, but for the sake of clarity,
+  the future discussion will center specifically on the case where the time
+  is determined by the validator nodes.
+- **Reliability**. The time value must be tolerant to the malicious behavior
+  of validator nodes.
+- **Agreement**. The time must be the same on all the nodes to ensure that
+  transactions are executed in a deterministic manner. This means that the time
+  should be written in the Exonum blockchain storage. Thus, the “current” time
+  will be changing in the same way on all nodes during execution of transactions,
+  including during nodes update.
+- **Sufficient accuracy**. The specified time should be fairly accurate.
+  In practice, an acceptable deviation is a few seconds.
+- **Monotony**. The time value should only increase. A pragmatic requirement,
+  which simplifies use of time when implementing the business logic.
+
+## Preliminaries
+
+Two possible approaches satisfying the above criteria are
+[integration with consensus](#integration-with-consensus)
+and [a time oracle service](#time-oracle-service).
+
+### Integration with Consensus
+
+!!! tip
+    Consult the [consensus description](../architecture/consensus.md) for
+    explanation of `Propose` and `Precommit` messages, and the “+2/3” notation.
+
+The validator includes its local time in each `Precommit` message.
+At the next height the leader includes +2/3 `Precommit`s of the previous block
+into the `Propose`. The time median of these `Precommit`s is recorded into
+the header of the block obtained based on this `Propose`.
+
+#### Advantages
+
+- The time value is indicated directly in the header of each block making it
+  more accessible.
+- Time is forcibly updated in the course of consensus operation:
+  it is impossible to sabotage update of time without stopping consensus.
+- Blockchain is not clogged by transactions associated with time determination.
+
+#### Disadvantages
+
+- The consensus code becomes more complex. (Time is included into the consensus
+  logic while [anchoring](bitcoin-anchoring.md) and
+  [configuration](configuration-updater.md) are not.)
+- Time updates are tied to creation of `Precommit` messages. In the case
+  of a large delay in block acceptance, all the transactions therein will
+  be executed with the same outdated time value.
+
+### Time Oracle Service
+
+Each validator at a specific time sends a transaction indicating its local time
+(usually immediately after the commit of each block). Exonum storage contains
+an index with the most current time indicated separately by each validator.
+A median of these index values is stored separately; it is considered the
```

Comment:
I assume 'contains an index' means 'contains some kind of Exonum table'. If that's true, I'd say 'A median of these ~~index~~values', because it's already clear where these values are stored and 'index values' sounds confusing (what do these values _index_?).